website/src/controllers/api-v3/user.js

```diff
@@ -31,4 +39,98 @@ api.getUser = {
   },
 };
 
+let partyMembersFields = 'profile.name stats achievements items.special';
+
+/**
+ * @api {post} /user/class/cast/:spell Cast a spell on a target.
+ * @apiVersion 3.0.0
+ * @apiName UserCast
+ * @apiGroup User
+ *
+ * @apiParam {string} spellId The spell to cast.
+ * @apiParam {string="party","self","user","task"} targetType Query parameter, the target type for the spell.
+ * @apiParam {UUID} targetId Optional query parameter, the id of the target when casting a spell on a party member or a task.
+ *
+ * @apiSuccess {Object|Array} mixed Will return the modified targets. For party members only the necessary fields will be populated.
+ */
+api.castSpell = {
+  method: 'POST',
+  middlewares: [authWithHeaders(), cron],
+  url: '/user/class/cast/:spell',
+  async handler (req, res) {
+    let user = res.locals.user;
+    let targetType = req.query.targetType;
+    let targetId = req.query.targetId;
+    let spellId = req.params.spellId;
+
+    // optional because not required by all targetTypes, presence is checked later if necessary
+    req.checkQuery('targetId', res.t('targetIdUUID')).optional().isUUID();
+    req.checkQuery('targetType', res.t('invalidTargetType')).isIn(['task', 'self', 'user', 'party']);
+
+    let reqValidationErrors = req.validationErrors();
+    if (reqValidationErrors) throw reqValidationErrors;
+
+    let klass = common.content.spells.special[spellId] ? 'special' : user.stats.class;
+    let spell = common.content.spells[klass][spellId];
+
+    if (!spell) throw new NotFound(res.t('spellNotFound', {spell: req.params.spellId}));
+    if (spell.mana > user.stats.mp) throw new BadRequest(res.t('notEnoughMana'));
+
+    if (targetType === 'task') {
+      if (!targetId) throw new BadRequest(res.t('targetIdUUID'));
+
+      // TODO what about challenge tasks? should casting be disabled on them?
```

Comment:
Skills shouldn't apply to challenge tasks because it makes it harder for challenge owners to judge winners.

We'll get repeated newbies questions about why they can't cast a skill on some tasks, but we'll just have to handle them. I think the challenge owners are more important here.

It would be nice if casting on a challenge task produced an informative message rather than just not working - that would avoid questions.

Note though that none of this would have to be done for v3 go-live. I'd be happy to see just the question marks removed from that comment and a note added about the error message.